web/client/components/geostory/contents/Text.jsx

```diff
@@ -7,15 +7,34 @@
  */
 import React from 'react';
 import editableText from './enhancers/editableText';
-import { withHandlers, compose, branch} from 'recompose';
-import { Modes } from '../../../utils/GeoStoryUtils';
+import { withHandlers, withProps, compose, branch} from 'recompose';
+import localizedProps from '../../misc/enhancers/localizedProps';
 
-const Text = ({ id, toggleEditing = () => {}, html }) => {
+import { Modes, SectionTypes, EMPTY_CONTENT } from '../../../utils/GeoStoryUtils';
+
+
+const Text = ({ placeholder, id, toggleEditing = () => {}, html, contentEditing, mode}) => {
     return (
-        <div
-            id={id}
-            onClick={() => toggleEditing(true, html)}
-            dangerouslySetInnerHTML={{ __html: html }} />
+        <div style={{
+            minWidth: "200px",
+            minHeight: "40px"
+        }}>
+            {   // content => render it
+                html !== EMPTY_CONTENT && <div
+                id={id}
+                onClick={() => toggleEditing(true, html)}
+                dangerouslySetInnerHTML={{ __html: html }} />}
+            {   // no content => render a placeholder
+                !contentEditing && (!html || html === EMPTY_CONTENT) && mode === Modes.EDIT &&
```

Comment:
in this case you propose to change one condition to another one, how it should simplify the composed condition ?

Reply:
!contentEditing is unuseful because placeholder is not there without that condition satisfied, so it can be removed. 

Ideally you should do things once in the right place. The "right place" depends on the abstraction you have. 

If you are handling the placeholder presence outside (depending on contentEditing) this component should only take care of placing the placeholder, when present, in case of empty content.

The simplification is in terms of concept :
 -  in case of empty text, replace with the placeholder
VS 
 - in case of empty text, replace with the placeholder, except in case that contentEditing is false

And in terms of involved variables (2 instead of 3)